swift/xcode-autobuilder/XcodeBuildRunner.cpp

```diff
@@ -51,7 +51,22 @@ static bool exec(const std::vector<std::string>& argv) {
   return true;
 }
 
-void buildTarget(Target& target, bool dryRun) {
+static bool run_build_command(const std::vector<std::string>& argv, bool dryRun) {
+  if (dryRun) {
+    std::cout << absl::StrJoin(argv, " ") << "\n";
+  } else {
+    if (!exec(argv)) {
+      DIAGNOSE_ERROR(buildCommandFailed,
+                     "`autobuild` failed to run the build command:\n\n```\n{}\n```",
+                     absl::StrJoin(argv, " "));
+      codeql::Log::flush();
```

Comment:
I'm not sure we need this flush here, now that eventually we will always reach the end of `main` where flushing is done.